src/pages/contract/verify/index.tsx

```diff
@@ -28,29 +29,38 @@ export default function VerifiedContract() {
     version: "",
     license: "",
   });
+  const [evmVersion, setEvmVersion] = useState(defaultDropdownValue);
 
   const { connection } = useNetwork();
   const router = useRouter();
 
   const networkQuery = !getEnvironment().isDefaultConnection(connection)
     ? { network: connection }
     : {};
+  const { data: verificationConfig } = useGetVerificationConfigQuery({
+    network: connection,
+  });
 
-  const evmVersions = [
-    { label: "default", value: "default" },
-    { label: "paris", value: "paris" },
-    { label: "london", value: "london" },
-    { label: "berlin", value: "berlin" },
-    { label: "istanbul", value: "istanbul" },
-    { label: "petersburg", value: "petersburg" },
-    { label: "constantinople", value: "constantinople" },
-    { label: "byzantium", value: "byzantium" },
-    { label: "spuriousDragon", value: "spuriousDragon" },
-    { label: "tangerineWhistle", value: "tangerineWhistle" },
-    { label: "homestead", value: "homestead" },
-  ];
+  const getCompilerVersions = (language) => {
+    let builds: string[];
+    if (language === ContractLanguage.Solidity) {
+      builds = verificationConfig?.solidity_compiler_versions ?? [];
+    } else {
+      builds = verificationConfig?.vyper_compiler_versions ?? [];
+    }
+    return builds.map((build) => ({
+      label: build,
+      value: build,
+    }));
+  };
```

Comment:
This would loop everytime the dropdown changes.

How about if prepare the arrays in `useMemo` and choose if sol | viper to have less computation
```[{
'sol': {label, value},
'vyper: {label: value}
}]```

Reply:
Updated PR! Didn't use `useMemo` but the result should be the same